src/world/footpath.c

```diff
@@ -1138,12 +1140,14 @@ static void loc_6A6D7E(
 					if (!ride_type_has_flag(ride->type, RIDE_TYPE_FLAG_FLAT_RIDE)) {
 						continue;
 					}
-					uint16 di = (mapElement->properties.track.sequence & 0x0F) | (mapElement->properties.track.type << 4);
-					if (!(RCT2_ADDRESS(0x0099CA64, uint8)[di] & (1 << 5))) {
+
+					const uint8 trackType = mapElement->properties.track.type;
+					const uint8 trackSequence = mapElement->properties.track.sequence & 0x0F;
+					if (!(FlatRideTrackSequenceProperties[trackType][trackSequence] & TRACK_SEQUENCE_FLAG_CONNECTS_TO_PATH)) {
 						return;
 					}
 					uint16 dx = ((direction - mapElement->type) & 3) ^ 2;
-					if (!(RCT2_ADDRESS(0x0099CA64, uint16)[di / 2] & (1 << dx))) {
+					if (!(FlatRideTrackSequenceProperties[trackType][trackSequence] & (1 << dx))) {
```

Comment:
I'm unsure regarding this change. It makes sense, but I'm not sure how it compares to the original for uneven trackSequences.